samza-core/src/main/java/org/apache/samza/operators/OperatorSpecGraph.java

```diff
@@ -54,20 +54,25 @@
   OperatorSpecGraph(StreamGraphSpec graphSpec) {
     this.inputOperators = graphSpec.getInputOperators();
     this.outputStreams = graphSpec.getOutputStreams();
+    this.broadcastStreams = graphSpec.getBroadcastStreams();
     this.tables = graphSpec.getTables();
     this.allOpSpecs = Collections.unmodifiableSet(this.findAllOperatorSpecs());
     this.hasWindowOrJoins = checkWindowOrJoins();
     this.serializedOpSpecGraph = opSpecGraphSerde.toBytes(this);
   }
 
-  public Map<StreamSpec, InputOperatorSpec> getInputOperators() {
+  public Map<String, InputOperatorSpec> getInputOperators() {
     return inputOperators;
   }
 
-  public Map<StreamSpec, OutputStreamImpl> getOutputStreams() {
+  public Map<String, OutputStreamImpl> getOutputStreams() {
     return outputStreams;
   }
 
+  public Set<String> getBroadcastStreams() {
```

Comment:
Is the expectation here that StreamSpec has already been serialized to StreamConfig with the streamId as the key in the config object? Removing the StreamSpec object from OperatorSpecGraph would mean that the information included in StreamSpec (which is currently a serializable object) would need to be in some serialized format stored externally somewhere. Also, if we make this change, ideally, we should remove TableSpec reference from this class as well. I am wondering what's the specific reason for removing the reference to StreamSpec here?